### DISCUSSION TITLE
Bot saved in database

### DISCUSSION THREAD

**Discussion by elitekiril** [Zrips/CMI#7287] [Q&A] [answered]
So recently i've got bot attacks and they got saved into database and are now seen in /baltop even tho they never came into server and nothing was showed on console about them since its protected with an anti bot system.

**Comment by TomLewis** [comment, marked as answer]
I too noticed this, its so weird, as no other plugin saves the bots, There are no Essential files created for them, but its so weird that CMI does.

@kiriles do you also use CMI on a proxy?

**Reply by elitekiril** [reply]
Actually im only using it on my survival only but the thing is i just noticed that it didn't save them if i use a proper anti bot protection, if you need help i can help you out via discord. Kiky#9483

**Reply by mrfloris** [reply]
CMI has some bungee features https://www.zrips.net/cmi/bungeecord/

When someone has successfully joined, they basically get logged by certain plugins, including CMI. However, some plugins recognize join/leave with no playtime and have no new data to log. CMI however still has to register logout position and balance value and some other data. 

Anyway, you can use `/cmi removeuser [uuid/duplicates]` command to purge a player from the database safely.
